File: examples/validation/test_with.py
# Test with statement support
def read_file(filename: str) -> str:
    """Read file using with statement."""
    with open(filename) as f:
        content = f.read()
    return content

def write_file(filename: str, content: str) -> None:
    """Write file using with statement."""
    with open(filename, "w") as f:
        f.write(content)

def process_file(input_file: str, output_file: str) -> None:
    """Process file with multiple with statements."""
    with open(input_file) as fin:
        data = fin.read()

    with open(output_file, "w") as fout:
        fout.write(data.upper())

File: examples/validation/test_with.rs
use std::io::Read;
use std::io::Write;
#[doc = "Read file using with statement."]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn read_file(filename: String) -> Result<String, std::io::Error> {
    let mut f = std::fs::File::open(&filename)?;
    let content = {
        let mut content = String::new();
        f.read_to_string(&mut content)?;
        content
    };
    Ok(content.to_string())
}
#[doc = "Write file using with statement."]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn write_file(filename: String, content: &str) -> Result<(), std::io::Error> {
    let mut f = std::fs::File::create(&filename)?;
    f.write_all(content.as_bytes()).unwrap();
    Ok(())
}
#[doc = "Process file with multiple with statements."]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn process_file(input_file: String, output_file: String) -> Result<(), std::io::Error> {
    let mut fin = std::fs::File::open(&input_file)?;
    let data = {
        let mut content = String::new();
        fin.read_to_string(&mut content)?;
        content
    };
    let mut fout = std::fs::File::create(output_file.as_ref().unwrap())?;
    fout.write_all(data.to_uppercase().as_bytes()).unwrap();
    Ok(())
}
